### ISSUE TITLE
Odd error message from attempted array concatenation 

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4944]
## tl;dr

If a user tries to concatenate arrays using a syntax familiar from other languages, they'd see this error message that seems difficult to interpret.

```
$ echo '{a:[1,2],b:[3,4]}' | zq -Z 'yield a+b' -
error("type ID too large for primitive: 30")
```

## Details

Repro is with Zed commit 59b72d8.

In Python for example, array concatenation can look like:

```
$ python3
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> a=[1,2]
>>> b=[3,4]
>>> print(a+b)
[1, 2, 3, 4]
```

As the Zed docs point out, array concatenation is currently performed with the [spread operator](https://zed.brimdata.io/docs/language/expressions#array-expressions).

```
$ zq -version
Version: v1.11.1-27-g59b72d88

$ echo '{a:[1,2],b:[3,4]}' | zq -z 'yield [...a,...b]' -
[1,2,3,4]
```

Perhaps at some point we'd want to support the `+` syntax as well. But if it's something something we intend to do right away, we might want to improve the error message.

**Comment by philrz** [comment]
Verified in Zed commit 38763f8.

Revisiting the original repro example, the error message is much improved.

```
$ zq -version
Version: v1.14.0-16-g38763f82

$ echo '{a:[1,2],b:[3,4]}' | zq -Z 'yield a+b' -
error("type [int64] incompatible with '+' operator")
```

Thanks @mattnibs!
